servant-server/src/Servant/Server/Internal.hs

```diff
@@ -10,11 +10,13 @@
 {-# LANGUAGE ScopedTypeVariables        #-}
 {-# LANGUAGE TypeFamilies               #-}
 {-# LANGUAGE TypeOperators              #-}
+{-# LANGUAGE UndecidableInstances       #-}
```

Comment:
Not sure how contentious this is. I needed to use this to get around the whole "Constraint is no smaller than instance head". The full error is here:

```
Internal.hs|458 col 10 error| Constraint is no smaller than the instance head                                                                                                                 
||   in the constraint: HasConfigEntry       
||                        config (BasicAuthCheck (AuthReturnType (BasicAuth realm)))
|| (Use UndecidableInstances to permit this) 
|| In the instance declaration for           
||   ‘HasServer (BasicAuth realm :> api) config’
```